src/main/java/org/springframework/hateoas/core/JsonPathLinkDiscoverer.java

```diff
@@ -178,6 +178,16 @@ private JsonPath getExpression(String rel) {
 	 */
 	@Override
 	public boolean supports(MediaType delimiter) {
-		return this.mediaType == null ? true : this.mediaType.isCompatibleWith(delimiter);
+
+		if (this.mediaTypes == null) {
```

Comment:
This by definition will never happen, I think we have to check for an empty collection here.